shared/src/api/extension/flatExtensionApi.ts

```diff
@@ -103,6 +134,18 @@ export const initNewExtensionAPI = (
                     )
                 )
             ),
+
+        // Language
+        getHover: (textParameters: TextDocumentPositionParams) => {
+            const document = textDocuments.get(textParameters.textDocument.uri)
+
+            const invokeProvider = (
+                provider: sourcegraph.HoverProvider
+            ): sourcegraph.ProviderResult<sourcegraph.Badged<Hover>> =>
+                provider.provideHover(document, toPosition(textParameters.position))
```

Comment:
It doesn't seem necessary to create a new `Position` instance every time we invoke a provider?

```suggestion
            const document = textDocuments.get(textParameters.textDocument.uri)
            const position = toPosition(textParameters.position)

            const invokeProvider = (
                provider: sourcegraph.HoverProvider
            ): sourcegraph.ProviderResult<sourcegraph.Badged<Hover>> =>
                provider.provideHover(document, position)
```